### PR TITLE
Added the ability to use functions as plugin params

### PR DESCRIPTION
**Pull request by jableader**
- Added missing submodules
- Made it so you can pass in functions directly instead of this name business
  + If you pass in a string there is a deprication warning, and it will try to Get the value from the runtime. Not quite backwards compatable but better than nothing
  + I suspect I'm logging incorrectly in twitchyliquid64/CNC/utl/functions.go

### CODE REVIEW

File: src/github.com/twitchyliquid64/CNC/plugin/exec/structure.go [hunk 1]
```diff
@@ -23,8 +23,21 @@ type Hook interface{
...
+
+func (self JSInvocation) Call() error {
+  var thisObject otto.Value;
+  if (self.ThisObject != nil) {
+    thisObject = self.ThisObject.Value();
+  } else {
+    thisObject = otto.UndefinedValue();
+  }
+
+  _, err := self.Callback.Call(thisObject, self.Parameters...)
```
Comment (twitchyliquid64) [inline src/github.com/twitchyliquid64/CNC/plugin/exec/structure.go]: change thisObject type declaration to interface{}, and set to nil in the else statement.

File: src/github.com/twitchyliquid64/CNC/util/functions.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "github.com/robertkrimen/otto"
+)
+
+func GetFunc(argument otto.Value, vm *otto.Otto) otto.Value {
+  if (argument.IsFunction()) {
+    return argument;
+  }
+
+  mname := argument.String()
+  logging.Warning("Using a string to name parameters is now depricated. Please use callbacks directly: " + mname)
```
Comment (twitchyliquid64) [inline src/github.com/twitchyliquid64/CNC/util/functions.go]: logging.Warning takes two parameters: also fix spelling

 try this:

logging.Warning("plugin-builtin", "Using a string to name parameters is now deprecated. Please use callbacks directly: " + mname)

File: src/github.com/twitchyliquid64/CNC/util/functions.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  method, err := vm.Get(mname)
+  if (err != nil){
+    logging.Error("Error getting function \"" + mname + "\": ")
```
Comment (twitchyliquid64) [inline src/github.com/twitchyliquid64/CNC/util/functions.go]: Likewise, two parameters.

File: src/github.com/twitchyliquid64/CNC/plugin/exec/structure.go (hunk 1 above)
Comment (jableader) [inline src/github.com/twitchyliquid64/CNC/plugin/exec/structure.go]: Value.Call accepts another Value only, cant pass in nil
https://godoc.org/github.com/robertkrimen/otto#Value.Call

**Pull request merged**